### PR TITLE
feat: add `types` subpackage export

### LINKED ISSUE #26: feat: add types to exports
### Clear and concise description of the problem

`resolveCommand` is really awesome for CLI apps but also can be really useful for building `snippet` blocks in front end applications when you want the user to be able to choose their package manager.

The problem is that they way this package is structured you must be in an environment that has access to **node**. So for instance in Cloudflare Pages there is no access to **node** so when using `resolveCommand` even if we aren't using `detect` it will fail when deploying because **node** doesn't exist in that environment.

### Suggested solution

Would changing the way you import things make a difference to this?

The ideal structure would add `package-manager-detector/types` to the exports in `package.json` so that you could have all the types without having to load `index.ts` and therefore all modules.

This way you would import from `package-manager-detector/types` and `package-manager-detector/commands` and that would hopefully avoid needing `node:fs` since `detect.ts` would not be loaded.
 
Example:
```ts
- import { ... } from "package-manager-detector" -
+ import { ... } from "package-manager-detector/types" +
+ import { ... } from "package-manager-detector/commands" +
```

### Alternative

_No response_

### Additional context

It's pretty simple to prove/disprove this we would just need to add:

**package.json**:
```jsonc
{
	// ...
	"exports": {
		// ...
		"./types": {
	      "import": "./dist/types.mjs",
	      "require": "./dist/types.cjs"
	    }
	}
}
```

and

**build.config.ts**:
```ts
import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  entries: [
    'src/index.ts',
    'src/commands.ts',
    'src/detect.ts',
    'src/constants.ts',
	// +++
    'src/types.ts',
	// +++
  ],
  clean: true,
  declaration: true,
  rollup: {
    emitCJS: true,
    inlineDependencies: true,
  },
})
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

Willing to open a PR for this if you think this is a good idea!

### PR DESCRIPTION
**Pull request by userquin**
### Description

This PR adds the types moduke to the unbuild command and the entries to the package.json.

### Linked Issues
resolves #26

### CODE REVIEW

File: build.config.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import { defineBuildConfig } from 'unbuild'
 
-export default defineBuildConfig({
+export default defineBuildConfig([{
```
Comment (userquin) [inline build.config.ts:3]: rmeove the array

**Comment by benmccann** [comment]
I'm not much of a Typescript expert, but is this the right thing to do? I can't remember using any other packages where you import types from a separate subpath and am not even sure how that would work.

If I understand correctly what they want to do is `import { resolveCommand } from 'package-manager-detector/commands'`. But how would you import it both from there and from `'package-manager-detector/commands'`?

I think the issue might be that there is no `types` field under `commands` in the `package.json`

**Comment by userquin** [comment]
I guess it shouldn't use the default package to import `resolveCommand`, this code:
```
import {
	resolveCommand,
	type Agent,
	type Command,
	type ResolvedCommand,
} from "package-manager-detector";
```

should be:
```ts
import type { Agent Command, ResolvedCommand } from "package-manager-detector";
import { resolveCommand } from "package-manager-detector/commands";
```

**Review by antfu — APPROVED** [review]

**Review by benmccann — CHANGES_REQUESTED** [review]
I think we can close this per @userquin's last comment

**Comment by antfu** [comment]
Either way would work for me (I often expose `/types` for quite a few packages), but yeah, to the original issue is more like a usage issue

**Pull request closed without merging**